### PR TITLE
feat(es7): using new elasticsearch client

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |no|
|New feature?   |yes|	
|BC breaks?     |no|	
|Deprecations?  |no|	
|Fixed tickets  |no|	

The goal of this PR is using a new elasticsearch client from the common. All dependencies to the real elasticsearch library should be removed. **Important this PR will not break anything**!

The new client development on the commonBundle: 
https://github.com/ems-project/EMSCommonBundle/pull/74/files

### CODE REVIEW

File: Controller/ElasticsearchController.php [hunk 1]
```diff
@@ -158,26 +141,25 @@ public function statusAction($_format)
...
 
-            if ('html' === $_format && 'green' !== $status['status']) {
-                $globalStatus = $status['status'];
-                if ('red' === $status['status']) {
+            if ('html' === $_format && !$health->isGreen()) {
+                $globalStatus = $health->getStatus();
+                if ('red' === $globalStatus) {
                     $this->getLogger()->error('log.elasticsearch.cluster_red', [
-                        'color_status' => $status['status'],
+                        'color_status' => $health['status'],
```
Comment (dameert) [inline Controller/ElasticsearchController.php:148]:
```suggestion
                        'color_status' => $globalStatus,
```

File: Controller/ElasticsearchController.php [hunk 2]
```diff
@@ -158,26 +141,25 @@ public function statusAction($_format)
...
                     ]);
                 } else {
                     $this->getLogger()->warning('log.elasticsearch.cluster_yellow', [
-                        'color_status' => $status['status'],
+                        'color_status' => $health['status'],
```
Comment (dameert) [inline Controller/ElasticsearchController.php:152]:
```suggestion
                        'color_status' => $globalStatus,
```

Comment (Davidmattei) [reply Controller/ElasticsearchController.php]: Good catch fixed

File: Controller/ElasticsearchController.php (hunk 1 above)
Comment (Davidmattei) [reply Controller/ElasticsearchController.php]: Good catch fixed

File: Command/RevisionCopyCommand.php [hunk 3]
```diff
@@ -92,10 +92,9 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $input->getArgument(self::ARG_JSON_MERGE) ?? ''
         );
 
-        $request = $copyContext->makeRequest();
-        $request->setSize((int) $input->getOption(self::OPTION_BULK_SIZE));
+        $size = (int) $input->getOption(self::OPTION_BULK_SIZE);
 
-        foreach ($this->elasticsearchService->scroll($request) as $i => $response) {
+        foreach ($this->client->scroll($copyContext->getIndex(), $copyContext->getBody(), $size) as $i => $response) {
```
Comment (dameert) [inline Command/RevisionCopyCommand.php:97]: I like the refactor!

File: Controller/ContentManagement/NatureController.php [hunk 4]
```diff
@@ -65,27 +65,25 @@ public function reorderAction(ContentType $contentType, Request $request, Conten
...
-                'sort' => $contentType->getOrderField(),
-            ]
-        ]);
-
-        if ($result['hits']['total'] > $this::MAX_ELEM) {
+        $search = $this->elasticsearchClient->searchByContentType(
+            $contentType->getEnvironment()->getAlias(),
+            $contentType->getName(),
+            ['sort' => $contentType->getOrderField()],
+            self::MAX_ELEM
```
Comment (dameert) [inline Controller/ContentManagement/NatureController.php:72]: This line solves a potential bug in the existing code 👍

**Comment by dameert** [comment]
Lol at automation

File: Controller/Views/CriteriaController.php [hunk 5]
```diff
@@ -674,13 +671,15 @@ public function addCriteriaRevision(View $view, array $rawData, $targetFieldName
             ]);
 
             return $revision;
-        } else if ($result['hits']['total'] == 1) {
+        } else if ($searchResponse->getTotal() == 1) {
             /**@var Revision $revision*/
             $revision = null;
-            if (isset($loadedRevision[$result['hits']['hits'][0]['_id']])) {
-                $revision = $loadedRevision[$result['hits']['hits'][0]['_id']];
+            $firstId = $searchResponse->getDocumentCollection()->first()->getId();
```
Comment (dameert) [inline Controller/Views/CriteriaController.php:677]: can't call getId() on nullable (first might return null). However you are kind of checking this. Is phpstan complaining or not locally on this line?

File: Controller/ElasticsearchController.php [hunk 6]
```diff
@@ -559,12 +559,8 @@ public function searchApiAction(Request $request)
             //http://blog.alterphp.com/2012/08/how-to-deal-with-asynchronous-request.html
             $request->getSession()->save();
 
-            /** @var Client $client */
-            $client = $this->getElasticsearch();
-
             $this->getLogger()->debug('Before search api');
-            $results = $client->search($params);
-
+            $results = $this->elasticsearchClient->searchByRequest($searchRequest)->toArray();
```
Comment (dameert) [inline Controller/ElasticsearchController.php:563]: should you not use the $body variable in the search?

File: Controller/ElasticsearchController.php [hunk 7]
```diff
@@ -749,21 +749,17 @@ public function searchAction(Request $request)
...
-                    }
+
+            foreach ($this->elasticsearchClient->getAliases() as $alias) {
+                if (!isset($environments[$alias->getName()])) {
+                    continue;
+                }
+
+                $mapAlias[$environments[$alias->getName()]['alias']] = $environments[$alias->getName()];
+                foreach ($alias->getIndexes() as $index) {
+                    $mapIndex[$index] = $environments[$alias->getName()];
```
Comment (dameert) [inline Controller/ElasticsearchController.php:762]: Trusting you on this one :p

File: Controller/ElasticsearchController.php (hunk 6 above)
Comment (Davidmattei) [reply Controller/ElasticsearchController.php:563]: The body is in the request object that we are passing?

File: Controller/ElasticsearchController.php (hunk 7 above)
Comment (Davidmattei) [reply Controller/ElasticsearchController.php:762]: I tested this so no worries

File: Controller/ElasticsearchController.php (hunk 6 above)
Comment (dameert) [reply Controller/ElasticsearchController.php:563]: I dont see the line where this happens?

Comment (dameert) [reply Controller/ElasticsearchController.php:563]: checked it again, where you create the body variable, until this line. no set is happening on the SearchRequest

File: Controller/ElasticsearchController.php [hunk 8]
```diff
@@ -863,11 +820,10 @@ public function searchAction(Request $request)
                 $body['aggs']['agg_' . $id] = $option->getConfigDecoded();
             }
 
-
-            $params['body'] = $body;
+            $searchRequest->setBody($body);
```
Comment (dameert) [inline Controller/ElasticsearchController.php:823]: Here you are setting the body, but this is "too late" for the other call! and we are in another function :p

File: Controller/ElasticsearchController.php (hunk 6 above)
Comment (Davidmattei) [reply Controller/ElasticsearchController.php:563]: Omg you're right sorry! Will fix it

Comment (Davidmattei) [reply Controller/ElasticsearchController.php:564]: Fixed

File: Controller/Views/CriteriaController.php (hunk 5 above)
Comment (Davidmattei) [reply Controller/Views/CriteriaController.php:677]: Also fixed

**Pull request merged**